src/db.ts

```diff
@@ -0,0 +1,65 @@
+import fsExtra from 'fs-extra';
+const { mkdtempSync, readFileSync, writeFileSync } = fsExtra;
```

Comment:
Was gonna ask about this - I haven't used ESM with TS and node very much. When I played with it there were some issues around importing CJS libs IIRC.  I didn't know if import-then-destructure was necessary here or just preference.

Reply:
It was necessary due to the issues you mentioned: when loading a CJS lib from ESM in node, node exposes the exports object as the default export.

I'm not sure if using native ESM is the right choice.  I thought it might be nice to learn, but if you want to go back to CJS I'm cool with that.

Reply:
Ah, I thought it was personal preference. I assumed you'd use `ts-node` :)

My best experience so far with ESM was writing JS/jsdoc and using `tsc` as a linter. That was pleasant, but for a tiny project.

Also using vite now for a project, but that's totally transparent AFAICT.  I have no strong feelings. Would imagine if it gets painful it's easy to switch.